OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/FullscreenDialogFragment.kt

```diff
@@ -0,0 +1,136 @@
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = DialogFullscreenLoaderBinding.inflate(inflater)
+        isCancelable = false
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, args: Bundle?) {
+        super.onViewCreated(view, args)
+        intiViews()
+        initObservers()
+        if (iapViewModel.isVerificationPending) {
+            iapViewModel.executeOrder()
+        }
+    }
+
+    private fun intiViews() {
+        binding.materialTextView.setText(getTitle(), TextView.BufferType.SPANNABLE)
+    }
+
+    private fun initObservers() {
+        iapViewModel.errorMessage.observe(viewLifecycleOwner, NonNullObserver { errorMsg ->
+            if (errorMsg.throwable is InAppPurchasesException) {
+                when (errorMsg.throwable.httpErrorCode) {
+                    HttpStatus.UNAUTHORIZED -> {
+                        environment.router?.forceLogout(
+                            requireContext(),
+                            environment.analyticsRegistry,
+                            environment.notificationDelegate
+                        )
+                        return@NonNullObserver
+                    }
+                    else -> showUpgradeErrorDialog(errorMsg.errorResId)
+                }
+            } else {
+                showUpgradeErrorDialog(errorMsg.errorResId)
+            }
+            iapViewModel.errorMessageShown()
+        })
+    }
+
+    private fun showUpgradeErrorDialog(
+        @StringRes errorResId: Int = R.string.general_error_message
+    ) {
+        AlertDialogFragment.newInstance(
+            getString(R.string.title_upgrade_error),
+            getString(errorResId),
+            getString(R.string.label_close),
+            { _, _ -> resetPurchase() },
+            getString(R.string.label_get_help),
+            { _, _ ->
+                environment.router?.showFeedbackScreen(
+                    requireActivity(),
+                    getString(R.string.email_subject_upgrade_error)
+                )
+                resetPurchase()
+            }).show(childFragmentManager, null)
+    }
+
+    private fun getTitle(): SpannableStringBuilder {
+        val unlocking = getString(R.string.fullscreen_loader_unlocking)
+        val fullAccess = getString(R.string.fullscreen_loader_full_access)
+        val toYourCourse = getString(R.string.fullscreen_loader_to_your_course)
+
+        val spannable =
+            SpannableStringBuilder(String.format("%s\n%s\n%s", unlocking, fullAccess, toYourCourse))
+
+        spannable.setSpan(
+            ForegroundColorSpan(ContextCompat.getColor(requireContext(), R.color.accentAColor)),
+            unlocking.length,
+            unlocking.length + fullAccess.length + 1,
+            Spannable.SPAN_EXCLUSIVE_INCLUSIVE
+        )
+        return spannable
+    }
+
+    private fun resetPurchase() {
+        iapViewModel.reset()
```

Comment:
IMO, shouldn't we call `iapViewModel.reset` right after execute API is failed rather than resetting it on dialog button click?

Reply:
The viewModel `reset` function sets `isVerificationPending` to true which will call `execute` API if user rotate his screen with error dialog.